### ISSUE TITLE
关于您论文中的D.F deformation fields形变图的绘制

### ISSUE THREAD

**Issue by 2805413893** [Kid-Liet/Reg-GAN#20]
尊敬得作者您好，看到您文章中使用了D.F形变图，想请教您一下是如何绘制的

**Comment by Kid-Liet** [comment]
您好，参考CycTrainer.py文件中的这个函数即可

![image](https://user-images.githubusercontent.com/52225942/175472581-94f8c538-5767-4394-8b1b-fc50d546e9da.png)

**Comment by 2805413893** [comment]
非常感谢您的回复，但是我还是有一个小疑惑，您这个函数里的defms参数输入的是fake_B的numpy数组吗？还是其他什么参数？这里有点疑问，感谢！

**Comment by Kid-Liet** [comment]
您好，defms就是形变场，不是fakeB，我并没有把它额外放入测试代码中，如果您想加入可以参考如下改写：


 #############val###############
            with torch.no_grad():
                MAE = 0
                num = 0
                for i, batch in enumerate(self.val_data):
                    real_A = Variable(self.input_A.copy_(batch['A']))
                    real_B = Variable(self.input_B.copy_(batch['B']))
                    fake_B = self.netG_A2B(real_A)
                    Trans = self.R_A(fake_B,real_B) .squeeze()    #(B C W H) 这里默认batch为1，去掉第一维度，配合写形变场
                    self.save_deformation(self,Trans ,root="xxxx")
                    mae = self.MAE(fake_B.detach().cpu().numpy().squeeze(),real_B.detach().cpu().numpy().squeeze())
                    MAE += mae
                    num += 1

                print ('Val MAE:',MAE/num)

**Comment by 2805413893** [comment]
非常感谢您的帮助，已经得到形变图了，十分感谢
